Code/ItemRandoComponent.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+
+namespace ArchipelagoRandomizer
+{
+	public abstract class ItemRandoComponent : MonoBehaviour
+	{
+		/// <summary>
+		/// Preloads any objects this component needs to reference
+		/// </summary>
+		public virtual void Preload(Preloader preloader)
+		{
+			//
+		}
+
+		/// <summary>
+		/// This component is enabled when a file is started or loaded while the randomizer is active
+		/// </summary>
+		protected virtual void OnEnable()
+		{
+			//
+		}
+
+		/// <summary>
+		/// This component is disabled when the randomizer becomes inactive
+		/// </summary>
+		protected virtual void OnDisable()
+		{
+			//
+		}
+
+		private void Start()
+		{
+			ItemRandomizer.OnEnabled += OnEnable;
+			ItemRandomizer.OnDisabled += () =>
```

Comment:
Why does this not subscribe `OnDisable` to it? If you're wanting to ensure the `enabled = false` code is always run by default, remember overrides can use `base.OnDisable()` and most IDEs will insert that automatically.

Reply:
In my case, it wasn't automatically inserting that, so I did this to allow for minimal work for anyone creating an ItemRandoComponent

Though when I implement this and will change the way components are initialized, I'll probably have to change how ItemRando gets disabled, so I'll revisit this when that time comes.